### PR TITLE
Fix deprecation warnings from webpack 5

### PR DESCRIPTION
**Pull request by ronami**
### Summary

Hopefully the final PR to support webpack v5 without any warnings: This PR doesn't change any functionality but adjusts the plugin to webpack 5's deprecation warnings (creates a lot of noise).

### CODE REVIEW

File: src/lib/index.ts [hunk 1]
```diff
@@ -38,8 +38,15 @@ class TPAStylePlugin {
...
+    const pluginDescriptor = isWebpack5
+      ? TPAStylePlugin.pluginName
+      : {
+          name: TPAStylePlugin.pluginName,
+          stage: compilation.PROCESS_ASSETS_STAGE_OPTIMIZE,
+        };
+
     compiler.hooks.compilation.tap(TPAStylePlugin.pluginName, compilation => {
-      compilation.hooks.optimizeChunkAssets.tapAsync(TPAStylePlugin.pluginName, (chunks, callback) => {
+      compilation.hooks.processAssets.tapAsync(pluginDescriptor, (chunks, callback) => {
```
Comment (ronami) [inline src/lib/index.ts]: `optimizeChunkAssets` has been deprecated and will be removed in v6. This keeps the same functionality on v5 and above.

File: src/lib/index.ts [hunk 2]
```diff
@@ -38,8 +38,15 @@ class TPAStylePlugin {
     );
     this.replaceRuntimeModule(compiler);
 
+    const pluginDescriptor = isWebpack5
+      ? TPAStylePlugin.pluginName
+      : {
+          name: TPAStylePlugin.pluginName,
+          stage: compilation.PROCESS_ASSETS_STAGE_OPTIMIZE,
+        };
```
Comment (ronami) [inline src/lib/index.ts]: This new hook now requires a `stage` enum.

Comment (ronami) [reply src/lib/index.ts]: This is similar to how its done in other plugins: https://github.com/NMFR/last-call-webpack-plugin/pull/9/files.

File: src/lib/index.ts [hunk 3]
```diff
@@ -76,9 +83,12 @@ class TPAStylePlugin {
   private extract(compilation, chunks) {
     const promises = [];
 
+    const files = isWebpack5 ? [...chunk.files] : chunk.files;
```
Comment (ronami) [inline src/lib/index.ts]: This has been changed from an array to a Set.

File: src/lib/index.ts [hunk 4]
```diff
@@ -39,7 +39,16 @@ class TPAStylePlugin {
     this.replaceRuntimeModule(compiler);
 
     compiler.hooks.compilation.tap(TPAStylePlugin.pluginName, compilation => {
-      compilation.hooks.optimizeChunkAssets.tapAsync(TPAStylePlugin.pluginName, (chunks, callback) => {
+      const pluginDescriptor = isWebpack5
+        ? TPAStylePlugin.pluginName
```
Comment (ronnyrin) [inline src/lib/index.ts]: are you sure it's not the other way around?

Comment (ronami) [reply src/lib/index.ts]: Oh, is is the other way around, nice catch, fixing.

**Pull request merged**